parser/parser.py

```diff
@@ -145,6 +151,66 @@ def p_statement_if(p):
     'statement : IF expression THEN statement ELSE THEN statement END_OF_IF_THEN_ELSE END_OF_STATEMENT'
     p[0] = ('if-statement', p[2], p[4], p[7])
 
+def p_variable_parameters(p):
+    'parameters : NAME LIST_ITEM_SEPARATOR parameters'
+    p[0] = ('parameters', p[1], p[3])
+
+def p_variable_parameters_base(p):
+    '''parameters : NAME
+                  | empty'''
+    p[0] = ('parameters', p[1])
+
+def p_empty(p):
+    'empty :'
+    p[0] = None
+
+'''
+----------------- functions start ----------------- bare for oversikt
+'''
+def p_function_assign(p):
```

Comment:
Do we need all these cases?
I think all the return things can be dropped if you define a return statement as 
statement : RETURN expression END_OF_STATEMENT. Then a return can happen anywhere in the function body, including at the end, and we do not need to explicitly have the return in each grammar case.